allennlp/models/encoder_decoders/simple_seq2seq.py

```diff
@@ -148,9 +148,7 @@ def forward(self,  # type: ignore
                 if timestep == 0:
                     # For the first timestep, when we do not have targets, we input start symbols.
                     # (batch_size,)
-                    with torch.cuda.device_of(final_encoder_output):
-                        input_choices = Variable(torch.LongTensor([self._start_index]).expand_as(
-                                final_encoder_output[:, 0]))
+                    input_choices = Variable(source_mask.data.new().resize_(batch_size).fill_(self._start_index))
```

Comment:
Is there a way to write a test to see if this creates tensors on the same device?

Reply:
The existing tests show that this works on the CPU, and we know that this works on the GPU also.  We have a hard time automatically testing things on GPUs, though, which is why this bug made it through in the first place.

Reply:
Okay. This comment seems to indicate this should work: https://github.com/pytorch/pytorch/blob/master/torch/tensor.py#L13-L15

I didn't go through the CPP code to understand what's happening though.

Reply:
This is the recommended way to do these operations: https://discuss.pytorch.org/t/is-it-possible-to-make-a-function-transparent-between-cpu-and-gpu/1235/7.  Though apparently we missed that you pass the shape directly to `new`...